packages/@uppy/locales/src/en_US.js

```diff
@@ -1,9 +1,10 @@
+/* eslint-disable comma-dangle */
 const en_US = {}
 
 en_US.strings = {
   addBulkFilesFailed: {
     '0': 'Failed to add %{smart_count} file due to an internal error',
-    '1': 'Failed to add %{smart_count} files due to internal errors',
+    '1': 'Failed to add %{smart_count} files due to internal errors'
```

Comment:
Unfortunately not, unless you know of a small lib that does?

Reply:
This seems to do the trick:

```js
import { Linter } from 'eslint'
const linter = new Linter
const formattedLocale = linter
  .verifyAndFix("var json=" + JSON.stringify(combinedLocale), {
    rules: {
      "comma-dangle": ["error", "always"],
      "quote-props": ["error", "as-needed"],
      "object-property-newline": "error",
      "object-curly-newline": ["error", "always"],
      "key-spacing": ["error", { mode: "strict" }],
      indent: ["error", 2],
    },
  })
  .output.replace(/^var\s+json\s*=\s*/, "");
```

Reply:
See my point above:

> Eslint was used to fix linter issues but it significantly slowed down the process. Instead I'm ignoring the comma dangle rule in the locale files, which was the only thing it was fixing.

Reply:
Maybe we can mesure that, I'm pretty sure `Linter` is significantly faster than `stringify-object`+`ESLint`.

Reply:
It is a generated file but also the file that people would use as a template when starting a new translation, so it would be best if it does have correct formatting

Reply:
We _could_ do that, but is it worth the effort? If this was part of our codebase I would say so but this is essentially a dist file, just a build output we don't touch or really look at. I didn't measure it but as soon as I added fix step it felt a lot slower. So I think we are fine here with the eslint ignore comment. What do you think?

Reply:
What I think is we should make the extra-effort of adding trailing commas :) How often do we run this script anyway? How big is the difference really? Also my suggestion allows us to get rid of `stringify-object` which is nice I think. I won't die on that hill, but that's my opinion since that's what you are asking.

Reply:
```console
$ time corepack yarn workspace locale-pack build
✅ Generated '/Users/duhamean/Documents/transloadit/uppy/packages/@uppy/locales/src/en_US.js'
✅ Generated locale docs
✅ Generated types
corepack yarn workspace locale-pack build  5.06s user 0.74s system 127% cpu 4.543 total
$ git reset --hard
HEAD is now at 7a65f5e99 Run script to add locale to newly added `screen-capture.md`
$ time corepack yarn workspace locale-pack build
✅ Generated '/Users/duhamean/Documents/transloadit/uppy/packages/@uppy/locales/src/en_US.js'
✅ Generated locale docs
✅ Generated types
corepack yarn workspace locale-pack build  4.54s user 0.70s system 139% cpu 3.753 total
```

The difference seems to be about half a second, so ~10%.

Reply:
The slow one I was referring to is `ESLint`, wasn't sure about `Linter`. Still two problems.

Your code snippet does not work unless I recreate out eslint config in `private/locale-pack` (it can't find our root eslint packages and settings), which seems a bit much. I could just add the parser and add the rules manually but if we change the rules then we must also think of changing it there. 

```
  fixed: false,
  messages: [
    {
      ruleId: null,
      fatal: true,
      severity: 2,
      message: "Parsing error: The keyword 'const' is reserved",
      line: 2,
      column: 1
    }
  ],
```

Reply:
The `Linter` class is not supposed to interact with the FS at all, it would be very surprising to have to recreate the eslint config (and not worth it I agree). I was able to run my code snippet without changing any config file, should we pair on this to sort this out?

Reply:
Took an easier approach, fixed in https://github.com/transloadit/uppy/pull/3276/commits/de8c74b2b7198f2b67c2aeaf374d95d57c17c538

Reply:
Let's remove stringify-object from the dependencies and we're good.

Reply:
We can but then all the keys would become strings, which would be inconsistent with the other locale files.

```diff
- addingMoreFiles: 'Adding more files',
+ 'addingMoreFiles': 'Adding more files',
```

Reply:
We can but then all keys would have quotes, which is inconsistent with other locale files.

[quote-props](https://eslint.org/docs/rules/quote-props) with `eslint --fix` won't solve this as we ignore this rule for locale files.

```diff
- addingMoreFiles: 'Adding more files',
+ 'addingMoreFiles': 'Adding more files',
```

Reply:
Yeah let's not have quotes, it's good for now. I may try to remove the dependency and reactivate the rule for those files in a follow-ip PR (unless I stumble upon a blocking point), but it should not block the current PR to land.